packages/web-component/src/lib/descope-wc/DescopeWc.ts

```diff
@@ -365,20 +395,32 @@ class DescopeWc extends BaseDescopeWc {
       htmlUrl: getContentUrl(projectId, `${readyScreenId}.html`),
       htmlLocaleUrl:
         filenameWithLocale && getContentUrl(projectId, filenameWithLocale),
+      samlIdpUsername,
     };
 
     const lastAuth = getLastAuth(loginId);
 
     // If there is a start screen id, next action should start the flow
-    // But if oidcIdpStateId is not empty, this optimization doesn't happen
-    if (showFirstScreenOnExecutionInit(startScreenId, oidcIdpStateId)) {
+    // But if oidcIdpStateId, samlIdpStateId, ssoAppId is not empty, this optimization doesn't happen
```

Comment:
Can we add the reason for skipping this optimization so it will be clear for the next developer that's working on this area?

Reply:
Added. in general it because there are scenarios that we may decide not to show the first screen (in cases like a user is already logged in) mostly in SSO scenarios.